libtiledbsoma/test/common.cc

```diff
@@ -46,12 +46,46 @@ static std::unique_ptr<ArrowSchema> _create_col_info_schema(
 //   array-creation error.)
 const int CORE_DOMAIN_MAX = 2147483646;
 
+// Notes:
+//
+// * This is multi-purpose code used for generic SOMASparseNDArray,
+//   SOMADenseNDArray, and SOMADataFrame
+// * The NDArrays always have:
+//   o int64 dims soma_dim_0, soma_dim_1, ..., soma_dim_n
+//   o a single attr of varying numeric type -- float32, float32, int16, etc.
+// * Default-indexed (i.e. what almost everyone uses) SOMADataFrame has
+//   o int64 dims soma_joinid
+//   o arbitrary number of attrs of arbitrary type -- int, float, bool, string,
+//     categorical-of-string, you name it
+// * But SOMADataFrame, in the SOMA spec, just needs to have soma_joinid
+//   present, as a dim or attr.
+//   o soma_joinid can be a dim, along with others
+//   o soma_joinid can be an attr only, with something else being the dim
+//
+// Given this context, create_arrow_schema_and_index_columns is a factory
+// for creating schema information for ND arrays and default-indexed
+// dataframes.
+//
+// * This returns a pair of ArrowSchema and ArrowTable
+// * ArrowTable, in turn, is a pair of ArrowArray and ArrowSchema
+// * So this is
+//    o ArrowSchema -- schema for the array
+//      - ArrowArray  -- information about the tile extent, domain, and
+//        current domain for the index columns
+//      - ArrowSchema -- information about the datatypes for the index columns
+//    o Confusingly, there's even more nesting: for an n-attr dataframe, there
+//      is an ArrowSchema with its n_children being n -- and each of those
+//      n entries in its children[] array are also of type ArrowSchema
+// * The data structures conform to arrow_adapter
+// * The Python and R bindings prepare similar Arrow information when
+//   passing a create-array request to libtiledbsoma.
+
 std::pair<std::unique_ptr<ArrowSchema>, ArrowTable>
 create_arrow_schema_and_index_columns(
     int64_t dim_max, bool use_current_domain) {
-    // Create ArrowSchema for SOMAArray
+    // Create ArrowSchema for the entire SOMAArray: dims and attrs both
     auto arrow_schema = std::make_unique<ArrowSchema>();
-    arrow_schema->format = "+s";
+    arrow_schema->format = "+s";  // non-leaf node
```

Comment:
nit: Was this comment meant for `arrow_schema->n_children = 2;` below? For other lines you have it listed for `n_children` and not the `format`.

Reply:
yeah that could go either way ... `"+s"` for struct-type, or, `n_children` > 0 ... I'll make it consistent though -- thanks!